### PR TITLE
GUVNOR-2308: Project explorer minimize/maximize button overlaps with breadcrumbs

### PR DESCRIPTION
**Pull request by paulovmr**
When the project explorer is wide, it looks like that:
image1

and when it is narrow, it looks like that:
image2

### CODE REVIEW

**Comment by manstis** [comment]
(Unrelated) the drop downs in the breadcrumbs look wrong - they have an icon "down arrow" and what appears to be a lower case "v".. we should only have the icon IIRC.

**Comment by paulovmr** [comment]
@manstis Yes, I don't know why but since I updated my repos this morning the workbench style became a little different (which includes the dropdowns).

**Comment by manstis** [comment]
Is it possible you can follow that up with Cristiano? He may have some idea
what changed in Uberfire.

**Comment by paulovmr** [comment]
Sure, I will talk to him (and investigate if this is only happening in my environment).

**Comment by paulovmr** [comment]
jenkins, retest this

**Comment by jhrcek** [comment]
@manstis @paulovmr The problem with changed dropdown styles is actually not local to Paulo. I'm downloading fresh 6.4.x snapshot every morning and the problem appeared only yesterday.

**Comment by cristianonicolai** [comment]
@paulovmr @jhrcek @manstis the style problem should be fixed by this PR: https://github.com/uberfire/uberfire-extensions/pull/178

**Comment by paulovmr** [comment]
@cristianonicolai Thanks!

I updated the images on this PR's description with screenshots containing the correct style, and changed a little the behavior of the expand/collapse button.

File: kie-wb-common-screens/kie-wb-common-project-explorer/kie-wb-common-project-explorer-client/src/main/java/org/kie/workbench/common/screens/explorer/client/widgets/navigator/Explorer.java [hunk 1]
```diff
@@ -238,6 +211,58 @@ public void onClick( final ClickEvent clickEvent ) {
         }
     }
 
+    private void addDivToAlignComponents() {
+        FlowPanel divClear = new FlowPanel();
+        divClear.getElement().getStyle().setClear( Style.Clear.BOTH );
+        container.add( divClear );
+    }
+
+    private void setupNavigatorExpandCollapseButton() {
```
Comment (Rikkola) [inline kie-wb-common-screens/kie-wb-common-project-explorer/kie-wb-common-project-explorer-client/src/main/java/org/kie/workbench/common/screens/explorer/client/widgets/navigator/Explorer.java]: Good move on making this into it's own method. 
I have been lately making classes to reduce the over all class lengths. In this case you could make a NavigatorButton or NavigatorExpandCollapseButton and put everything from lines 221-225 and 239-243 there. Leaving just the buttton.addClickHandler(...) here.

Comment (paulovmr) [inline kie-wb-common-screens/kie-wb-common-project-explorer/kie-wb-common-project-explorer-client/src/main/java/org/kie/workbench/common/screens/explorer/client/widgets/navigator/Explorer.java]: @Rikkola Good suggestion. I extended it a little and moved more stuff to this new class. Please, take a look and tell me what you think. :)

Comment (Rikkola) [inline kie-wb-common-screens/kie-wb-common-project-explorer/kie-wb-common-project-explorer-client/src/main/java/org/kie/workbench/common/screens/explorer/client/widgets/navigator/Explorer.java]: Looks good. Now what the button is and how it functions is separated. 
Just wondering why the method name is build(), but it does not build anything?

File: kie-wb-common-screens/kie-wb-common-project-explorer/kie-wb-common-project-explorer-client/src/main/java/org/kie/workbench/common/screens/explorer/client/widgets/navigator/NavigatorExpandCollapseButton.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+    private Button button;
+
+    public NavigatorExpandCollapseButton( final Mode mode ) {
+        this.mode = mode;
+        this.button = GWT.create( Button.class );
+        initWidget( button );
+    }
+
+    public void build( final ClickHandler clickHandler ) {
```
Comment (Rikkola) [inline kie-wb-common-screens/kie-wb-common-project-explorer/kie-wb-common-project-explorer-client/src/main/java/org/kie/workbench/common/screens/explorer/client/widgets/navigator/NavigatorExpandCollapseButton.java]: @paulovmr This is the build method that does not build. Just to be clear.

File: kie-wb-common-screens/kie-wb-common-project-explorer/kie-wb-common-project-explorer-client/src/main/java/org/kie/workbench/common/screens/explorer/client/widgets/navigator/Explorer.java (hunk 1 above)
Comment (paulovmr) [inline kie-wb-common-screens/kie-wb-common-project-explorer/kie-wb-common-project-explorer-client/src/main/java/org/kie/workbench/common/screens/explorer/client/widgets/navigator/Explorer.java]: @Rikkola I agree, already changed it. How about now?

**Comment by csadilek** [comment]
+1

**Comment by csadilek** [comment]
master: 1f0ca32d4804569715d6cda50eb167dc986f820f

**Pull request closed without merging**